### ISSUE TITLE
I use the CrossEntropyLoss to calculate the lcls,but lcls is much larger than lobj and lbox

### ISSUE THREAD

**Issue by gef1998** [ultralytics/yolov5#3403]
Will such a large lcls affect the back propagation of lobj and lbox
I made the following changes to employ  CrossEntropyLoss, I don't know if it is correct

loss.py#L129-l132
```Python
           CELcls = nn.CrossEntropyLoss()
           if model.nc > 1:  # cls loss (only if multiple classes)
                t = torch.full_like(ps[:, 5:], cn, device=device)  # targets
                t[range(n), tcls[i]] = cp
                #lcls += BCEcls(ps[:, 5:], t)  # BCE
                lcls += CELcls(ps[:, 5:], tcls[i].clone().detach()) 
```

yolo.py#L48-L59
```Python
if not self.training:  # inference
                if self.grid[i].shape[2:4] != x[i].shape[2:4]:
                    self.grid[i] = self._make_grid(nx, ny).to(x[i].device)

                y = x[i].sigmoid()
                y2 = x[i][...,5:]
                y2 = y2.softmax(dim=-1)
                y[..., 0:2] = (y[..., 0:2] * 2. - 0.5 + self.grid[i].to(x[i].device)) * self.stride[i]  # xy
                y[..., 2:4] = (y[..., 2:4] * 2) ** 2 * self.anchor_grid[i]  # wh
                y[...,5:] = y2
                z.append(y.view(bs, -1, self.no))
        return x if self.training else (torch.cat(z, 1), x)
```

![loss](https://user-images.githubusercontent.com/80207800/120303801-c3a9c980-c301-11eb-8557-3896b2d3992f.png)

**Comment by glenn-jocher** [comment]
@gef1998 we abandoned CELoss after experimenting with it in https://github.com/ultralytics/yolov3

The PyTorch CE loss function uses a slightly different averaging than the BCE loss function, so you may need to update your normalization technique to compensate. 

As a general principal if you want each of your losses to contribute equally they should be of similar magnitude.

**Comment by gef1998** [comment]
> 
> 
> @gef1998 we abandoned CELoss after experimenting with it in https://github.com/ultralytics/yolov3
> 
> The PyTorch CE loss function uses a slightly different averaging than the BCE loss function, so you may need to update your normalization technique to compensate.
> 
> As a general principal if you want each of your losses to contribute equally they should be of similar magnitude.

Thank you for your quickly answer!
Can I fix it by turning down CLS Loss Gain ?

**Comment by glenn-jocher** [comment]
@gef1998 yeah of course. It looks like you might want to reduce your cls loss gain by 100x. If you are training COCO this might correspond to an 80x reduction for your 80 classes.

**Comment by gef1998** [comment]
> @gef1998 yeah of course. It looks like you might want to reduce your cls loss gain by 100x. If you are training COCO this might correspond to an 80x reduction for your 80 classes.

Thank you very much！
